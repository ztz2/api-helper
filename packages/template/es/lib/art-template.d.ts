import artTemplate from './utils/template/art-template-web.js';
export default artTemplate;
