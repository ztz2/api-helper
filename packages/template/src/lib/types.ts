export interface ChangeCase {
  /**
   * @example
   * changeCase.camelCase('test string') // => 'testString'
   */
  camelCase: (value: string) => string
  /**
   * @example
   * changeCase.constantCase('test string') // => 'TEST_STRING'
   */
  constantCase: (value: string) => string
  /**
   * @example
   * changeCase.dotCase('test string') // => 'test.string'
   */
  dotCase: (value: string) => string
  /**
   * @example
   * changeCase.headerCase('test string') // => 'Test-String'
   */
  headerCase: (value: string) => string
  /**
   * @example
   * changeCase.lowerCase('TEST STRING') // => 'test string'
   */
  lowerCase: (value: string) => string
  /**
   * @example
   * changeCase.lowerCaseFirst('TEST') // => 'tEST'
   */
  lowerCaseFirst: (value: string) => string
  /**
   * @example
   * changeCase.paramCase('test string') // => 'test-string'
   */
  paramCase: (value: string) => string
  /**
   * @example
   * changeCase.pascalCase('test string') // => 'TestString'
   */
  pascalCase: (value: string) => string
  /**
   * @example
   * changeCase.pathCase('test string') // => 'test/string'
   */
  pathCase: (value: string) => string
  /**
   * @example
   * changeCase.sentenceCase('testString') // => 'Test string'
   */
  sentenceCase: (value: string) => string
  /**
   * @example
   * changeCase.snakeCase('test string') // => 'test_string'
   */
  snakeCase: (value: string) => string
  /**
   * @example
   * changeCase.swapCase('Test String') // => 'tEST sTRING'
   */
  swapCase: (value: string) => string
  /**
   * @example
   * changeCase.titleCase('a simple test') // => 'A Simple Test'
   */
  titleCase: (value: string) => string
  /**
   * @example
   * changeCase.upperCase('test string') // => 'TEST STRING'
   */
  upperCase: (value: string) => string
  /**
   * @example
   * changeCase.upperCaseFirst('test') // => 'Test'
   */
  upperCaseFirst: (value: string) => string
}
