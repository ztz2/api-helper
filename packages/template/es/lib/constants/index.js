export var API_CUSTOM_TEMPLATE_ID = 'API自定义分组ID_ea18930468d9402e';
export var MODEL_CUSTOM_TEMPLATE_ID = '模型自定义分组ID_6f3f32d6b404aaf9';
export var DEFAULT_SELECT_API_TPL_ID = 'typescript_api_76ea205dba9f3dfb';
export var DEFAULT_SELECT_MODEL_TPL_ID = 'javascript_map_841d8ffc9b34d1d5';
