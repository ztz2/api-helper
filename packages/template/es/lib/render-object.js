var __values = (this && this.__values) || function(o) {
    var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
    if (m) return m.call(o);
    if (o && typeof o.length === "number") return {
        next: function () {
            if (o && i >= o.length) o = void 0;
            return { value: o && o[i++], done: !o };
        }
    };
    throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
};
import merge from 'lodash/merge';
import _changeCase from 'change-case';
import cloneDeep from 'lodash/cloneDeep';
import * as changeCase from 'change-case';
import { filterSchemaPrimitiveValue } from '@api-helper/core/lib/utils/util';
import { postCode } from '../lib/utils/util';
export function renderObject(schema, api, options) {
    var _a, _b;
    options = merge({
        prefix: 'export ',
        suffixName: 'entity',
        onlyBody: false,
        paramType: 'request',
        emptyBodyCode: '{}',
    }, options);
    schema = cloneDeep(schema);
    var primitiveValueSchema = filterSchemaPrimitiveValue(schema);
    var prefix = options.prefix, suffixName = options.suffixName, dropComment = options.dropComment, emptyBodyCode = options.emptyBodyCode, _c = options.paramType, paramType = _c === void 0 ? 'request' : _c;
    // @ts-ignore
    var onlyBody = (_a = options === null || options === void 0 ? void 0 : options.onlyMap) !== null && _a !== void 0 ? _a : options.onlyBody;
    var keyword = "".concat(prefix, " const");
    var onRenderObjectName = (options === null || options === void 0 ? void 0 : options.onRenderObjectName) ? options.onRenderObjectName : renderObjectName;
    var commentCode = onlyBody ? '' : dropComment !== true ? renderObjectComment(api, paramType) : '';
    var objectName = (options === null || options === void 0 ? void 0 : options.name) ? options.name : onRenderObjectName(api, {
        suffixName: suffixName,
        paramType: paramType,
        changeCase: changeCase
    });
    /**
     * output ->  export class
     */
    var ki = ["".concat(keyword, " ").concat(objectName, " = ")].filter(Boolean).join('\n');
    var bodyCode;
    if (!schema || ((_b = schema === null || schema === void 0 ? void 0 : schema.params) === null || _b === void 0 ? void 0 : _b.length) === 0) {
        bodyCode = emptyBodyCode;
    }
    else {
        bodyCode = renderObjectDeepObject(primitiveValueSchema, undefined, true);
    }
    var code = postCode({
        ki: ki,
        commentCode: commentCode,
        code: bodyCode
    }, { onlyBody: onlyBody });
    // 移除最后的分号。
    if (onlyBody) {
        code = code.trim();
        if (code.endsWith(';')) {
            code = code.slice(0, code.length - 1);
        }
    }
    return code;
}
function renderObjectDeepObject(schema, memo, isRoot) {
    var e_1, _a;
    var _b;
    if (memo === void 0) { memo = new Map(); }
    if (isRoot === void 0) { isRoot = false; }
    if (memo.has(schema)) {
        return memo.get(schema);
    }
    memo.set(schema, 'null');
    if (isRoot) {
        debugger;
    }
    var codeWrap = [];
    var prefix = (schema === null || schema === void 0 ? void 0 : schema.type) === 'array' ? '[\n' : '{\n';
    var postfix = (schema === null || schema === void 0 ? void 0 : schema.type) === 'array' ? '\n]' : '\n}';
    try {
        for (var _c = __values(schema.params), _d = _c.next(); !_d.done; _d = _c.next()) {
            var child = _d.value;
            var keyName = (_b = child.keyName) !== null && _b !== void 0 ? _b : '';
            var type = child.type;
            var temporaryCode = [renderComment(child)];
            var v = "''";
            switch (type) {
                // 数组类型 | 对象类型
                case 'array':
                case 'object':
                    temporaryCode.pop();
                    temporaryCode.push(renderObjectDeepObject(child));
                    break;
                case 'string':
                    if ('enum' in child && child.enum.length > 0) {
                        v = "'".concat(child.enum[0], "'");
                    }
                    temporaryCode.push("".concat(keyName, ": ").concat(v));
                    break;
                case 'number':
                    temporaryCode.push("".concat(keyName, ": 0"));
                    break;
                case 'boolean':
                    temporaryCode.push("".concat(keyName, ": false"));
                    break;
                // 其他类型
                default:
                    temporaryCode.push("".concat(keyName, ": null"));
            }
            codeWrap.push(temporaryCode.join('\n'));
        }
    }
    catch (e_1_1) { e_1 = { error: e_1_1 }; }
    finally {
        try {
            if (_d && !_d.done && (_a = _c.return)) _a.call(_c);
        }
        finally { if (e_1) throw e_1.error; }
    }
    if (isRoot) {
        debugger;
    }
    var code = prefix + codeWrap.join(',\n') + postfix;
    if (schema === null || schema === void 0 ? void 0 : schema.keyName) {
        code = [
            renderComment(schema),
            "".concat(schema.keyName, ": ").concat(code)
        ].join('\n');
    }
    return code;
}
export function renderComment(schema) {
    var type = schema.type;
    if ('enum' in schema && schema.enum.length > 0) {
        type = schema.enum.map(function (item) { return "'".concat(item, "'"); }).join(' | ');
    }
    return "// { ".concat(type, " }").concat(schema.label ? " ".concat(schema.label) : '');
}
export function renderObjectName(api, options) {
    var _a, _b;
    var suffixName = (_a = options === null || options === void 0 ? void 0 : options.suffixName) !== null && _a !== void 0 ? _a : 'Entity ';
    var changeCase = (_b = options.changeCase) !== null && _b !== void 0 ? _b : _changeCase;
    var name = api.path;
    if (options.paramType) {
        name += " ".concat(options.paramType);
    }
    name += "".concat(suffixName, "By ").concat(api.method);
    return changeCase.camelCase(name);
}
function renderObjectComment(api, paramType, isExtraData) {
    if (isExtraData === void 0) { isExtraData = false; }
    var commentText = "/**\n * @description ".concat([api.title, api.description].filter(Boolean).join('、'), "\u3010").concat(isExtraData ? '不兼容的请求数据' : paramType === 'request' ? '请求数据' : paramType === 'response' ? '响应数据' : '', "\u5B9E\u4F53\u5BF9\u8C61\u3011").concat(api.docURL ? "\n * @doc ".concat(api.docURL) : '', "\n * @url [ ").concat(api.method.toUpperCase(), " ] ").concat(api.path, "\n */");
    return commentText;
}
