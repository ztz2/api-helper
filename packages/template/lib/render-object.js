"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __values = (this && this.__values) || function(o) {
    var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
    if (m) return m.call(o);
    if (o && typeof o.length === "number") return {
        next: function () {
            if (o && i >= o.length) o = void 0;
            return { value: o && o[i++], done: !o };
        }
    };
    throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.renderObjectName = exports.renderComment = exports.renderObject = void 0;
var merge_1 = __importDefault(require("lodash/merge"));
var cloneDeep_1 = __importDefault(require("lodash/cloneDeep"));
var changeCase = __importStar(require("change-case"));
var util_1 = require("@api-helper/core/lib/utils/util");
var util_2 = require("../lib/utils/util");
var change_case_1 = __importDefault(require("change-case"));
function renderObject(schema, api, options) {
    var _a, _b, _c;
    options = (0, merge_1.default)({
        prefix: 'export ',
        onlyBody: false,
        paramType: 'request',
        emptyBodyCode: '{}',
    }, options);
    schema = (0, cloneDeep_1.default)(schema);
    var schemaList = (0, util_1.filterSchemaPrimitiveValue)(Array.isArray(schema) ? schema : (_b = (_a = schema) === null || _a === void 0 ? void 0 : _a.params) !== null && _b !== void 0 ? _b : []);
    var prefix = options.prefix, dropComment = options.dropComment, emptyBodyCode = options.emptyBodyCode, _d = options.paramType, paramType = _d === void 0 ? 'request' : _d;
    // @ts-ignore
    var onlyBody = (_c = options === null || options === void 0 ? void 0 : options.onlyMap) !== null && _c !== void 0 ? _c : options.onlyBody;
    var keyword = "".concat(prefix, " const");
    var onRenderObjectName = (options === null || options === void 0 ? void 0 : options.onRenderObjectName) ? options.onRenderObjectName : renderObjectName;
    var commentCode = onlyBody ? '' : dropComment !== true ? renderObjectComment(api, paramType) : '';
    var objectName = (options === null || options === void 0 ? void 0 : options.name) ? options.name : onRenderObjectName(api, {
        paramType: paramType,
        changeCase: changeCase
    });
    /**
     * output ->  export class
     */
    var ki = ["".concat(keyword, " ").concat(objectName, " = ")].filter(Boolean).join('\n');
    var bodyCode;
    if (!schema || (Array.isArray(schema) && schema.length === 0)) {
        bodyCode = emptyBodyCode;
    }
    else {
        bodyCode = renderObjectDeepObject(schemaList);
    }
    var code = (0, util_2.postCode)({
        ki: ki,
        commentCode: commentCode,
        code: bodyCode
    }, { onlyBody: onlyBody });
    // 移除最后的分号。
    if (onlyBody) {
        code = code.trim();
        if (code.endsWith(';')) {
            code = code.slice(0, code.length - 1);
        }
    }
    return code;
}
exports.renderObject = renderObject;
function renderObjectDeepObject(schemaList, parentSchema, memo) {
    var e_1, _a;
    var _b;
    if (parentSchema === void 0) { parentSchema = null; }
    if (memo === void 0) { memo = new Map(); }
    if (memo.has(schemaList)) {
        return memo.get(schemaList);
    }
    memo.set(schemaList, 'null');
    var codeWrap = [];
    var prefix = (parentSchema === null || parentSchema === void 0 ? void 0 : parentSchema.type) === 'array' ? '[\n' : '{\n';
    var postfix = (parentSchema === null || parentSchema === void 0 ? void 0 : parentSchema.type) === 'array' ? '\n]' : '\n}';
    try {
        for (var schemaList_1 = __values(schemaList), schemaList_1_1 = schemaList_1.next(); !schemaList_1_1.done; schemaList_1_1 = schemaList_1.next()) {
            var schema = schemaList_1_1.value;
            var keyName = (_b = schema.keyName) !== null && _b !== void 0 ? _b : '';
            var type = schema.type;
            var temporaryCode = [renderComment(schema)];
            var v = "''";
            switch (type) {
                // 数组类型 | 对象类型
                case 'array':
                case 'object':
                    temporaryCode.pop();
                    temporaryCode.push(renderObjectDeepObject(schema.params, schema));
                    break;
                case 'string':
                    if ('enum' in schema && schema.enum.length > 0) {
                        v = "'".concat(schema.enum[0], "'");
                    }
                    temporaryCode.push("".concat(keyName, ": ").concat(v));
                    break;
                case 'number':
                    temporaryCode.push("".concat(keyName, ": 0"));
                    break;
                case 'boolean':
                    temporaryCode.push("".concat(keyName, ": false"));
                    break;
                // 其他类型
                default:
                    temporaryCode.push("".concat(keyName, ": null"));
            }
            codeWrap.push(temporaryCode.join('\n'));
        }
    }
    catch (e_1_1) { e_1 = { error: e_1_1 }; }
    finally {
        try {
            if (schemaList_1_1 && !schemaList_1_1.done && (_a = schemaList_1.return)) _a.call(schemaList_1);
        }
        finally { if (e_1) throw e_1.error; }
    }
    var code = prefix + codeWrap.join(',\n') + postfix;
    if (parentSchema === null || parentSchema === void 0 ? void 0 : parentSchema.keyName) {
        code = [
            renderComment(parentSchema),
            "".concat(parentSchema.keyName, ": ").concat(code)
        ].join('\n');
    }
    return code;
}
function renderComment(schema) {
    var type = schema.type;
    if ('enum' in schema && schema.enum.length > 0) {
        type = schema.enum.map(function (item) { return "'".concat(item, "'"); }).join(' | ');
    }
    return "// { ".concat(type, " }").concat(schema.label ? " ".concat(schema.label) : '');
}
exports.renderComment = renderComment;
function renderObjectName(api, options) {
    var _a;
    var changeCase = (_a = options.changeCase) !== null && _a !== void 0 ? _a : change_case_1.default;
    var name = api.path;
    if (options.paramType) {
        name += " ".concat(options.paramType);
    }
    name += "By ".concat(api.method);
    return changeCase.camelCase(name);
}
exports.renderObjectName = renderObjectName;
function renderObjectComment(api, paramType, isExtraData) {
    if (isExtraData === void 0) { isExtraData = false; }
    var commentText = "/**\n * @description ".concat([api.title, api.description].filter(Boolean).join('、'), "\u3010").concat(isExtraData ? '不兼容的请求数据' : paramType === 'request' ? '请求数据' : paramType === 'response' ? '响应数据' : '', "\u5B9E\u4F53\u5BF9\u8C61\u3011").concat(api.docURL ? "\n * @doc ".concat(api.docURL) : '', "\n * @url [ ").concat(api.method.toUpperCase(), " ] ").concat(api.path, "\n */");
    return commentText;
}
