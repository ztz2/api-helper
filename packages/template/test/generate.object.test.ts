import { join } from 'path';
import { APIHelper } from '@api-helper/core/es/lib/types';
import ParserSwagger from '@api-helper/core/es/lib/parser/parser-swagger';

import { readJsonFile } from './utils/file';
import { renderObject } from '../src/lib/render-object';

describe('生成 javascript 对象测试', () => {
  test('formatter', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-response-array.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[0];
    const api = category.apiList[0] as APIHelper.API;
    const code = renderObject(api.responseDataSchema, api, {
      onlyBody: true,
      paramType: 'response',
      formatter(schema) {
        const result = {useDefault: true, value: '""'};
        switch (schema.type) {
          case 'string':
            result.useDefault = false;
            result.value = '"字符串"';
            return result;
        }

        return result;
      },
    });
    expect(
      code,
    ).toMatchSnapshot('formatter');
  });

  test('formatter-null', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-response-array.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[0];
    const api = category.apiList[0] as APIHelper.API;
    const code = renderObject(null, api, {
      onlyBody: true,
      paramType: 'response',
      emptyBodyCode: 'null',
      formatter(schema) {
        const result = {useDefaults: true, value: ''};
        switch (schema.type) {
          case 'string':
            result.useDefaults = false;
            result.value = '"字符串"';
            return result;
        }
        return result;
      },
    });
    expect(
      code,
    ).toMatchSnapshot('formatter-null');
  });

  test('OpenAPI-2.0生成List对象', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-response-array.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[0];
    const api = category.apiList[0] as APIHelper.API;
    const code = renderObject(api.responseDataSchema, api, {
      paramType: 'response',
    });
    expect(
      code,
    ).toMatchSnapshot('OpenAPI-2.0生成List对象');
  });

  test('OpenAPI-2.0生成对象测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const api = category.apiList[0];
    const code = renderObject(api.responseDataSchema, api, {
      paramType: 'request',
      onlyBody: true,
    });
    expect(
      code,
    ).toMatchSnapshot('OpenAPI-2.0生成 interface 测试');
  });

  test('OpenAPI-3.0生成对象测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-3.0.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const api = category.apiList[1];
    const code = renderObject(api.requestDataSchema, api, {
      paramType: 'response',
    });
    expect(
      code,
    ).toMatchSnapshot('OpenAPI-3.00生成对象测试');
  });

  test('name - prefix - dropComment - onlyBody 属性测试', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const api = category.apiList[0];
    expect(
      renderObject(api.responseDataSchema, api, {
        paramType: 'request',
        name: '自定义名称',
      }),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 name测试');
    expect(
      renderObject(api.responseDataSchema, api, {
        paramType: 'request',
        prefix: 'export',
      }),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 prefix测试');
    expect(
      renderObject(api.responseDataSchema, api, {
        paramType: 'request',
        dropComment: true,
      }),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 dropComment测试');
    expect(
      renderObject(api.responseDataSchema, api, {
        paramType: 'request',
        onlyBody: true,
      }),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 onlyBody测试');
  });

  test('空属性', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[1];
    const api = category.apiList[0];
    expect(
      renderObject(null, api),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 null 属性');
    expect(
      renderObject({params: [], type: 'object'} as unknown as APIHelper.Schema, api),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 空数组');
  });

  test('array单参数', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0-dot.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[0];
    const api = category.apiList[0];
    expect(
      renderObject(api.requestDataSchema?.params?.find?.((item) => item.keyName === 'storeIds') ?? null, api),
    ).toMatchSnapshot('array单参数');
  });
  test('array单参数2', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0-dot.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[0];
    const api = category.apiList[0];
    expect(
      renderObject({
        'id': 'bc2f818f91624c8c',
        'type': 'array',
        'keyName': 'storeIds',
        'title': '',
        'description': '',
        'label': '',
        'rules': { 'required': true },
        'examples': [],
        'params': [],
        'enum': [],
      }, api),
    ).toMatchSnapshot('array单参数2');
  });
  test('open-api-2.0-dot', async () => {
    const openAPIDocumentJSON = await readJsonFile(join(__dirname, '../../core/test/resources/open-api-2.0-dot.json'));
    const openAPIDocumentList = await new ParserSwagger(false).parser([openAPIDocumentJSON]);
    const openAPIDocument = openAPIDocumentList[0];
    const category = openAPIDocument.categoryList[0];
    const api = category.apiList[0];
    expect(
      renderObject(api.requestDataSchema, api),
    ).toMatchSnapshot('OpenAPI-2.0生成 对象 null 属性');
    expect(
      renderObject({params: [], type: 'object'} as unknown as APIHelper.Schema, api),
    ).toMatchSnapshot('open-api-2.0-dot');
  });
});
