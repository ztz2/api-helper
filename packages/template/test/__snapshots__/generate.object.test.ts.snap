// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`生成 javascript 对象测试 OpenAPI-2.0生成List对象: OpenAPI-2.0生成List对象 1`] = `
"/**
 * @description 获取用户列表信息【响应数据实体对象】
 * @url [ GET ] /system/app/personnels-staff-roster/worker-home-detail/{userId}
 */
export const systemAppPersonnelsStaffRosterWorkerHomeDetailUserIdResponseentityByGet =
  [
    {
      // { string } 门店名称
      storename: '',
      // { string } 所属组织
      organizeid: '',
      // { string } 所属组织名称
      organizename: '',
      // { string } 门店展示图
      storepic: '',
      // { string } 负责人
      storeheaduserid: '',
      // { string } 负责人
      storeheadusername: '',
      // { string } 定位
      address: '',
      // { string } 是否禁用
      disabled: '',
      // { string } 经度
      longitude: '',
      // { string } 纬度
      latitude: '',
      // { number } 距离
      distance: 0,
      // { array } 用户id
      users: [
        {
          // { string } 用户id
          userid: '',
          // { string } 门店id
          storeid: '',
          // { string } 用户姓名
          username: '',
          // { string } 手机号
          mobilenum: '',
          // { string } 岗位名称
          position: '',
          // { string } 可排班标识
          shiftsmarks: '',
          // { string } 账号
          account: '',
          // { string } 岗位名称
          organizeName: ''
        }
      ],
      // { string } 主键
      id: ''
    }
  ];
"
`;

exports[`生成 javascript 对象测试 OpenAPI-2.0生成对象测试: OpenAPI-2.0生成 interface 测试 1`] = `
"{
  // { number }
  id: 0,
  // { number }
  petId: 0,
  // { number }
  quantity: 0,
  // { string }
  shipDate: '',
  // { 'placed' | 'approved' | 'delivered' } Order Status
  status: 'placed',
  // { array } Order Tag
  tag: [
    [
      {
        // { 'error' | 'waring' | 'info' }
        tagName: 'error'
      }
    ]
  ],
  // { boolean }
  complete: false
}"
`;

exports[`生成 javascript 对象测试 OpenAPI-3.0生成对象测试: OpenAPI-3.00生成对象测试 1`] = `
"/**
 * @description Place an order for a pet、Place a new order in the store【响应数据实体对象】
 * @url [ POST ] /api/v3/store/order
 */
export const apiV3StoreOrderResponseentityByPost = {
  // { number }
  id: 0,
  // { number }
  petId: 0,
  // { number }
  quantity: 0,
  // { string }
  shipDate: '',
  // { 'placed' | 'approved' | 'delivered' } Order Status
  status: 'placed',
  // { boolean }
  complete: false
};
"
`;

exports[`生成 javascript 对象测试 array单参数: array单参数 1`] = `
"/**
 * @description list【请求数据实体对象】
 * @url [ GET ] /system/user/list
 */
export const systemUserListRequestentityByGet = {
  // { array } 门店id
  storeIds: []
};
"
`;

exports[`生成 javascript 对象测试 array单参数2: array单参数2 1`] = `
"/**
 * @description list【请求数据实体对象】
 * @url [ GET ] /system/user/list
 */
export const systemUserListRequestentityByGet = {
  // { array }
  storeIds: []
};
"
`;

exports[`生成 javascript 对象测试 formatter: formatter 1`] = `
"[
    {
      // { string } 门店名称
      storename: '字符串',
      // { string } 所属组织
      organizeid: '字符串',
      // { string } 所属组织名称
      organizename: '字符串',
      // { string } 门店展示图
      storepic: '字符串',
      // { string } 负责人
      storeheaduserid: '字符串',
      // { string } 负责人
      storeheadusername: '字符串',
      // { string } 定位
      address: '字符串',
      // { string } 是否禁用
      disabled: '字符串',
      // { string } 经度
      longitude: '字符串',
      // { string } 纬度
      latitude: '字符串',
      // { number } 距离
      distance: 0,
      // { array } 用户id
      users: [
        {
          // { string } 用户id
          userid: '字符串',
          // { string } 门店id
          storeid: '字符串',
          // { string } 用户姓名
          username: '字符串',
          // { string } 手机号
          mobilenum: '字符串',
          // { string } 岗位名称
          position: '字符串',
          // { string } 可排班标识
          shiftsmarks: '字符串',
          // { string } 账号
          account: '字符串',
          // { string } 岗位名称
          organizeName: '字符串'
        }
      ],
      // { string } 主键
      id: '字符串'
    }
  ]"
`;

exports[`生成 javascript 对象测试 formatter-null: formatter-null 1`] = `"null"`;

exports[`生成 javascript 对象测试 name - prefix - dropComment - onlyBody 属性测试: OpenAPI-2.0生成 对象 dropComment测试 1`] = `
"export const v2StoreOrderRequestentityByPost = {
  // { number }
  id: 0,
  // { number }
  petId: 0,
  // { number }
  quantity: 0,
  // { string }
  shipDate: '',
  // { 'placed' | 'approved' | 'delivered' } Order Status
  status: 'placed',
  // { array } Order Tag
  tag: [
    [
      {
        // { 'error' | 'waring' | 'info' }
        tagName: 'error'
      }
    ]
  ],
  // { boolean }
  complete: false
};
"
`;

exports[`生成 javascript 对象测试 name - prefix - dropComment - onlyBody 属性测试: OpenAPI-2.0生成 对象 name测试 1`] = `
"/**
 * @description Place an order for a pet【请求数据实体对象】
 * @url [ POST ] /v2/store/order
 */
export const 自定义名称 = {
  // { number }
  id: 0,
  // { number }
  petId: 0,
  // { number }
  quantity: 0,
  // { string }
  shipDate: '',
  // { 'placed' | 'approved' | 'delivered' } Order Status
  status: 'placed',
  // { array } Order Tag
  tag: [
    [
      {
        // { 'error' | 'waring' | 'info' }
        tagName: 'error'
      }
    ]
  ],
  // { boolean }
  complete: false
};
"
`;

exports[`生成 javascript 对象测试 name - prefix - dropComment - onlyBody 属性测试: OpenAPI-2.0生成 对象 onlyBody测试 1`] = `
"{
  // { number }
  id: 0,
  // { number }
  petId: 0,
  // { number }
  quantity: 0,
  // { string }
  shipDate: '',
  // { 'placed' | 'approved' | 'delivered' } Order Status
  status: 'placed',
  // { array } Order Tag
  tag: [
    [
      {
        // { 'error' | 'waring' | 'info' }
        tagName: 'error'
      }
    ]
  ],
  // { boolean }
  complete: false
}"
`;

exports[`生成 javascript 对象测试 name - prefix - dropComment - onlyBody 属性测试: OpenAPI-2.0生成 对象 prefix测试 1`] = `
"/**
 * @description Place an order for a pet【请求数据实体对象】
 * @url [ POST ] /v2/store/order
 */
export const v2StoreOrderRequestentityByPost = {
  // { number }
  id: 0,
  // { number }
  petId: 0,
  // { number }
  quantity: 0,
  // { string }
  shipDate: '',
  // { 'placed' | 'approved' | 'delivered' } Order Status
  status: 'placed',
  // { array } Order Tag
  tag: [
    [
      {
        // { 'error' | 'waring' | 'info' }
        tagName: 'error'
      }
    ]
  ],
  // { boolean }
  complete: false
};
"
`;

exports[`生成 javascript 对象测试 open-api-2.0-dot: OpenAPI-2.0生成 对象 null 属性 1`] = `
"/**
 * @description list【请求数据实体对象】
 * @url [ GET ] /system/user/list
 */
export const systemUserListRequestentityByGet = {
  // { string } 创建者
  '2021-05': '',
  // { array } 门店id
  storeIds: [],
  // { string } 有分页参数的话就加,根据你的实际查询参数编写
  '有分页参数的话就加,根据你的实际查询参数编写': '',
  // { string } 创建者
  createBy: '',
  // { array } ID集合
  ids: [],
  // { array } 主键id集合
  pics: [],
  // { string } 创建时间
  createTime: '',
  // { string }
  ancestors: '',
  // { object }
  dept: {
    // { string } 创建者
    createBy: '',
    // { string }
    ancestors: ''
  },
  // { array }
  roles: [
    {
      // { string } 创建者
      createBy: '',
      // { string } 创建时间
      createTime: ''
    }
  ]
};
"
`;

exports[`生成 javascript 对象测试 open-api-2.0-dot: open-api-2.0-dot 1`] = `
"/**
 * @description list【请求数据实体对象】
 * @url [ GET ] /system/user/list
 */
export const systemUserListRequestentityByGet = {};
"
`;

exports[`生成 javascript 对象测试 空属性: OpenAPI-2.0生成 对象 null 属性 1`] = `
"/**
 * @description Place an order for a pet【请求数据实体对象】
 * @url [ POST ] /v2/store/order
 */
export const v2StoreOrderRequestentityByPost = {};
"
`;

exports[`生成 javascript 对象测试 空属性: OpenAPI-2.0生成 对象 空数组 1`] = `
"/**
 * @description Place an order for a pet【请求数据实体对象】
 * @url [ POST ] /v2/store/order
 */
export const v2StoreOrderRequestentityByPost = {};
"
`;
